### PR TITLE
WIP: Standardization/integration

### PR DESCRIPTION
**Pull request by AxelPeter**
Spinner (UU-639)
Checkbox (UU-639)
Navbar (UU-589)

Related to
https://github.com/ovh-ux/ovh-module-emailpro/pull/26
https://github.com/ovh-ux/ovh-module-exchange/pull/70
https://github.com/ovh-ux/ovh-module-office/pull/13
https://github.com/ovh-ux/ovh-module-sharepoint/pull/19
https://github.com/ovh-ux/ovh-angular-sidebar-menu/pull/12

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -75,14 +76,12 @@
...
-    "ovh-angular-manager-navbar": "~6.6.4",
     "ovh-angular-sidebar-menu": "^5.0.0",
-    "ovh-angular-toggleclass": "~1.2.0",
     "angular-translate": "^2.15.1",
     "ovh-manager-webfont": "^1.0.1",
     "ovh-angular-responsive-tabs": "^3.1.4",
-    "ovh-ui-angular": "^1.1.0",
-    "ovh-ui-kit": "https://github.com/ovh-ux/ovh-ui-kit.git#1.0.0-alpha.19",
+    "ovh-ui-angular": "git+https://github.com/ovh-ux/ovh-ui-angular#standardization/integration",
+    "ovh-ui-kit": "git+https://github.com/ovh-ux/ovh-ui-kit#fd923bf704c0f843839056a8e765323a6854113c",
```
Comment (antleblanc) [inline bower.json]: Quick note: `ovh-ui-kit` has to be released and then update here to the correct version.

File: bower.json [hunk 2]
```diff
@@ -112,7 +111,8 @@
     "ovh-angular-proxy-request": "^0.1.0",
     "angular-ui-router": "^0.4.2",
     "animate.css": "~3.5.0",
-    "ovh-api-services": "^2.1.9"
+    "ovh-api-services": "^2.1.9",
+    "ovh-ui-kit": "fd923bf704c0f843839056a8e765323a6854113c"
```
Comment (antleblanc) [inline bower.json]: Same as above.

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<!-- <div class="oui-navbar-menu oui-navbar-menu_fixed oui-navbar-menu_right"> -->
+    <header class="oui-navbar-menu__header">
+        <oui-back-button class="oui-navbar-menu__back"
+            on-click="$ctrl.toggleMenu($ctrl.menuName, true)">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: What about `data-` prefix?

File: client/app/components/manager-navbar/manager-navbar.html [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        <span class="oui-navbar-toggler__hamburger">
+            <span></span>
+            <span></span>
+            <span></span>
+            <span></span>
+        </span>
+    </button>
+    <div class="oui-navbar-menu oui-navbar-menu_fixed oui-navbar-menu_toggle">
+        <ul class="oui-navbar-list">
+            <li class="oui-navbar-list__item" ng-repeat="managerLink in $ctrl.navigation.managerLinks">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar.html]: `track by`?

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        </oui-back-button>
+        <span class="oui-navbar-menu__breadcrumb" ng-bind="::$ctrl.headerBreadcrumb"></span>
+        <h3 class="oui-navbar-menu__title" ng-bind="::$ctrl.headerTitle"></h3>
+    </header>
+    <ul class="oui-navbar-list">
+        <li class="oui-navbar-list__item" ng-repeat="menuLink in $ctrl.menuLinks">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: track by?

File: client/app/components/manager-navbar/manager-navbar.html [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    <a class="oui-navbar__brand"
+        ng-attr-aria-label="$ctrl.navigation.brand.label"
+        ng-href="{{::$ctrl.navigation.brand.url}}">
+        <img class="navbar-logo" src="assets/images/navbar/icon-logo-ovh.svg" alt="OVH">
+    </a>
+    <!-- /Brand -->
+
+    <!-- Managers -->
+    <ul class="oui-navbar-list oui-navbar_desktop-only">
+        <li class="oui-navbar-list__item" ng-repeat="managerLink in $ctrl.navigation.managerLinks">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar.html]: track by?

File: client/app/components/manager-navbar/manager-navbar.html [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                ng-href="{{::managerLink.url}}"
+                ng-bind="::managerLink.label">
+            </a>
+        </li>
+    </ul>
+    <!-- /Managers -->
+
+    <!-- Internal -->
+    <ul class="oui-navbar-list oui-navbar-list_end">
+        <li class="oui-navbar-list__item" ng-repeat="internalLink in $ctrl.navigation.internalLinks">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar.html]: track by?

File: client/app/components/manager-navbar/manager-navbar.html [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                <!-- /Menu Header for 'user' -->
+
+                <!-- General Menu Header -->
+                <header class="oui-navbar-menu__header oui-navbar_mobile-only" ng-if="internalLink.name !== 'user'">
+                    <h3 class="oui-navbar-menu__title" ng-bind="::internalLink.label"></h3>
+                </header>
+                <!-- General Menu Header -->
+
+                <ul class="oui-navbar-list">
+                    <li class="oui-navbar-list__item" ng-repeat="subLink in internalLink.subLinks">
```
Comment (antleblanc) [inline client/app/components/manager-navbar/manager-navbar.html]: track by?

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html (hunk 3 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: It's a custom html element, not necessary :)

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html (hunk 5 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: done for all ngRepeat :)

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            on-click="$ctrl.toggleMenu($ctrl.menuName, true)">
+        </oui-back-button>
+        <span class="oui-navbar-menu__breadcrumb" data-ng-bind="::$ctrl.headerBreadcrumb"></span>
+        <h3 class="oui-navbar-menu__title" data-ng-bind="::$ctrl.headerTitle"></h3>
+    </header>
+    <ul class="oui-navbar-list">
+        <li class="oui-navbar-list__item" data-ng-repeat="menuLink in $ctrl.menuLinks track by $index">
+            <a class="oui-navbar-link"
+                data-ng-if="::!menuLink.subLinks"
+                data-ng-click="$ctrl.toggleMenu()"
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: Another solution can be to listen the event `$stateChangeSuccess`

File: client/app/components/manager-navbar/manager-navbar.component.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+angular.module("App")
+    .component("managerNavbar", {
+        controller: "managerNavbarCtrl",
+        controllerAs: "$ctrl",
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.component.js]: Not needed

File: client/app/components/manager-navbar/manager-navbar.controller.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
+angular.module("App")
+    .controller("managerNavbarCtrl", function ($element, $document, $timeout, constants, translator, ManagerNavbarService) {
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.controller.js]: Class :)

File: client/app/components/manager-navbar/manager-navbar.controller.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+            // Get navbar navigation and user infos
+            ManagerNavbarService.getNavbar()
+                .then((navbar) => {
+                    this.currentUser = navbar.currentUser;
+                    this.navigation = navbar.navigation;
+                });
+        };
+
+        this.$postLink = () => {
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.controller.js]: Same as before :)

File: client/app/components/manager-navbar/manager-navbar.html [hunk 13]
```diff
@@ -0,0 +1,144 @@
+<!-- Navbar -->
+<nav class="oui-navbar oui-navbar_fixed" role="navigation">
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.html]: In the majority of cases setting an ARIA role and/or aria-* attribute that matches the default implicit ARIA semantics is unnecessary and not recommended as these properties are already set by the browser.

File: client/app/components/manager-navbar/manager-navbar.service.js [hunk 14]
```diff
@@ -0,0 +1,398 @@
+angular.module("App")
+    .service("ManagerNavbarService", function ($q, constants, translator, Products, User, LANGUAGES, OtrsPopupService, ssoAuthentication) {
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.service.js]: `Class`

File: client/app/components/manager-navbar/manager-navbar.service.js [hunk 15]
```diff
@@ -0,0 +1,398 @@
...
+        "use strict";
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.service.js]: --

File: client/app/components/manager-navbar/manager-navbar.service.js [hunk 16]
```diff
@@ -0,0 +1,398 @@
...
+            const currentLanguage = _(LANGUAGES).find({ value: translator.getLanguage() });
+            const currentLanguageName = _(currentLanguage).get("name");
+
+            return currentLanguageName;
+        };
+
+        const getProductsMenu = (categoryName, products) => {
+            const productsMenu = [];
+
+            _.forEach(products, (product) =>
```
Comment (jleveugle) [inline client/app/components/manager-navbar/manager-navbar.service.js]: _.map

File: client/app/components/manager-navbar/manager-navbar.html (hunk 13 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.html]: Ok, I take example from the bootstrap navbar for this part.
It's removed

File: client/app/components/manager-navbar/manager-navbar.service.js (hunk 14 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.service.js]: done

File: client/app/components/manager-navbar/manager-navbar.service.js (hunk 15 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.service.js]: done

File: client/app/components/manager-navbar/manager-navbar.service.js (hunk 16 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.service.js]: done

File: client/app/components/manager-navbar/manager-navbar.controller.js (hunk 11 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.controller.js]: done

File: client/app/components/manager-navbar/manager-navbar.component.js (hunk 10 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar.component.js]: done :)

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html (hunk 9 above)
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: For this case, yes.
But we have links with `target="_blank"`.

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html [hunk 17]
```diff
@@ -0,0 +1,37 @@
+<!-- <div class="oui-navbar-menu oui-navbar-menu_fixed oui-navbar-menu_right"> -->
+    <header class="oui-navbar-menu__header">
+        <oui-back-button class="oui-navbar-menu__back"
+            on-click="$ctrl.toggleMenu($ctrl.menuName, true)">
+        </oui-back-button>
+        <span class="oui-navbar-menu__breadcrumb" data-ng-bind="::$ctrl.headerBreadcrumb"></span>
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.html]: Only the `oui-back-button` is clickable in the header

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.controller.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+angular.module("App")
+    .controller("managerNavbarMenuCtrl", function ($element, ManagerNavbarService) {
+        "use strict";
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.controller.js]: done :)

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.controller.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
+angular.module("App")
+    .controller("managerNavbarMenuCtrl", function ($element, ManagerNavbarService) {
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.controller.js]: Yes, done :)

File: client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.component.js [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+    .component("managerNavbarMenu", {
+        bindings: {
+            headerTitle: "@?",
+            headerBreadcrumb: "@?",
+            menuName: "@",
+            menuLinks: "<",
+            navbarNav: "="
+        },
+        controller: "managerNavbarMenuCtrl",
+        controllerAs: "$ctrl",
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply client/app/components/manager-navbar/manager-navbar-menu/manager-navbar-menu.component.js]: Ok, I removed it :)

**Comment by AxelPeter** [comment]
The navbar will be added later. 
First, upgrade `ovh-ui-kit` and `ovh-ui-angular` to v2.0.0.

**Pull request closed without merging**